arrow-core/src/main/kotlin/arrow/core/extensions/validated/applicativeError/ValidatedApplicativeError.kt

```diff
@@ -0,0 +1,181 @@
+fun <E, A, B> Kind<Kind<ForValidated, E>, A>.redeem(
+  SE: Semigroup<E>,
+  arg1: Function1<E, B>,
+  arg2: Function1<A, B>
+): Validated<E, B> =
+  fix().redeem(arg1, arg2)
+
+@JvmName("attempt")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension kinded projected functions are deprecated", ReplaceWith("attempt()", "arrow.core.attempt"))
+fun <E, A> Kind<Kind<ForValidated, E>, A>.attempt(SE: Semigroup<E>): Validated<E, Either<E, A>> =
+  fix()._attempt()
+
+@JvmName("catch")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+// @Deprecated("@extension kinded projected functions are deprecated", ReplaceWith("Validated.catch(arg0, arg1)", "arrow.core.catch"))
+// TODO refactor suspend fun Validated.catch to inline fun Validated.catch, binary breaking
+fun <E, A> catch(
+  SE: Semigroup<E>,
+  arg0: Function1<Throwable, E>,
+  arg1: Function0<A>
+): Validated<E, A> =
+  try {
+    arg1.invoke().valid()
+  } catch (e: Throwable) {
+    arg0.invoke(e).invalid()
+  }
+
+@JvmName("catch")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+
+@Deprecated("This methods is invalid for Validated. ApplicativeError<F, Throwable> is inconsistent in `F`")
+fun <E, A> ApplicativeError<Kind<ForValidated, E>, Throwable>.catch(
+  SE: Semigroup<E>,
+  arg1: Function0<A>
+): Validated<E, A> = arrow.core.Validated.applicativeError<E>(SE).run {
+  this@catch.catch<A>(arg1) as arrow.core.Validated<E, A>
+}
+
+@JvmName("effectCatch")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+ @Deprecated("@extension kinded projected functions are deprecated", ReplaceWith("Validated.catch(arg1).mapLeft(arg0)", "arrow.core.catch"))
+suspend fun <E, A> effectCatch(
+  SE: Semigroup<E>,
+  arg0: Function1<Throwable, E>,
+  arg1: suspend () -> A
+): Validated<E, A> =
+  Validated.catch(arg1).mapLeft(arg0)
+
+@JvmName("effectCatch")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("This methods is invalid for Validated. ApplicativeError<F, Throwable> is inconsistent in `F`")
+suspend fun <E, F, A> ApplicativeError<F, Throwable>.effectCatch(
+  SE: Semigroup<E>,
+  arg1: suspend () -> A
+): Kind<F, A> = arrow.core.Validated.applicativeError<E>(SE).run {
+  this@effectCatch.effectCatch<F, A>(arg1) as arrow.Kind<F, A>
+}
+
+@Suppress(
+  "UNCHECKED_CAST",
+  "NOTHING_TO_INLINE"
+)
+@Deprecated("ApplicativeError typeclasses is deprecated. Use concrete methods on Validated")
+inline fun <E> Companion.applicativeError(SE: Semigroup<E>): ValidatedApplicativeError<E> = object :
+  arrow.core.extensions.ValidatedApplicativeError<E> {
+  override fun SE():
+    arrow.typeclasses.Semigroup<E> = SE
+}
```

Comment:
```suggestion
  ValidatedApplicativeError<E> { override fun SE(): Semigroup<E> = SE }
```

Reply:
I realised that this happens on all types, shall I create a separate branch with changes or we don't care that much as it's gonna go away anyway 😅